sdk/include/asm/trapamd64.inc

```diff
@@ -1,3 +1,54 @@
+/*
+ * PROJECT:     ReactOS SDK
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     ASM macros for x64 trap handling
+ * COPYRIGHT:   Copyright 2011-2024 Timo Kreuzer (timo.kreuzer@reactos.org)
+ */
+
+MACRO(ASSERT_TRAP_FRAME_INTS_ENABLED, Register)
+#if DBG
+    LOCAL IntsAreEnabled
+    test dword ptr [Register + KTRAP_FRAME_EFlags], HEX(200)
+    jnz IntsAreEnabled
+    int HEX(2C)
+IntsAreEnabled:
+#endif
+ENDM
+
+MACRO(ASSERT_TRAP_FRAME_IRQL_VALID, Register)
+#if DBG
+    LOCAL IrqlIsValid
+    mov rax, cr8
+    cmp byte ptr [Register + KTRAP_FRAME_PreviousIrql], al
+    je IrqlIsValid
+    int HEX(2C)
+IrqlIsValid:
+#endif
+ENDM
+
+MACRO(ASSERT_IRQL_PASSIVE)
+#if DBG
+    LOCAL IrqlIsPassive
+    mov rax, cr8
+    test rax, rax
+    jz IrqlIsPassive
+    int HEX(2C)
+IrqlIsPassive:
+#endif
+ENDM
+
+// Checks for user APCs and delvers them if necessary.
+// Clobbers all volatile registers except rax.
+MACRO(HANDLE_USER_APCS, ThreadReg, TrapFrame)
+    LOCAL NoUserApcPending
+
+    /* Check for pending user APC */
+    cmp byte ptr [ThreadReg + ThApcState + AsUserApcPending], 0
```

Comment:
Want to use `ThApiState + AsUserApcPending`, or `KTHREAD_UserApcPending` ? (even though both evaluate to the same thing)